### PR TITLE
Fix report_vuln/print_error in smb_ms17_010

### PR DESCRIPTION
**Pull request by wvu**
`report_vuln` was missing the necessary keys to store the vuln in the database. `print_error` got annoying when scanning a range of hosts.

- [x] `vulns` shows the vuln now
- [x] Scanning a range doesn't barf errors

#8167

### CODE REVIEW

File: modules/auxiliary/scanner/smb/smb_ms17_010.rb [hunk 1]
```diff
@@ -48,11 +48,10 @@ def run_host(ip)
...
         report_vuln(
-          :host   => rhost,
-          :port   => rport,
-          :proto  => 'tcp',
-          :sname  => 'SMB',
-          :info   => "Vulnerable to MS17-010",
+          host: ip,
+          name: self.name,
+          refs: self.references,
+          info: 'STATUS_INSUFF_SERVER_RESOURCES for FID 0 against IPC$'
```
Comment (wvu-r7) [inline modules/auxiliary/scanner/smb/smb_ms17_010.rb]: Happy to change this to something else, but this is what made the most sense to me, given that we're already reporting module title and references.

**Comment by jork2345** [comment]
Is it possible to make an exploit module of this vulnerability ?

**Comment by zerosum0x0** [comment]
jork2345 according to Microsoft, this is exploitable with EternalBlue, EternalRomance, and EternalSynergy

https://blogs.technet.microsoft.com/msrc/2017/04/14/protecting-customers-and-evaluating-risk/

These are some of the most complex exploits I've ever seen, and I don't say that lightly. Eternalromance does a ridiculous amount of "grooming", aka remote heap feng shui. In the case of EternalBlue, it simultaneously exploits SMBv1 and SMBv2, and talks Cairo, an undocumented SMB LanMan alternative.

tl;dr The exploit modules are gonna be huge, even if they simply just replay the pcaps of these exploits.

**Comment by tdoan-r7** [comment]
Running the vulns command with smb_ms17_010 now shows vulnerabilities and running the print_error command doesn't error out when scanning a range of hosts.
